Support newer Python and Robot Framework versions

vereszol opened the pull request:
Time has passed, this tool should be updated.

vereszol committed:
Drop support for Python 2 and 3.6

These versions are deprecated, dropping support to reduce
maintenance effort.

Pin down tox version to 3.x as syntax changed in 4.x; this can
be fixed later.

Pin down robotframework version as there are backwards incompatible
changes in versions 4 and newer; support for newer versions will
be added later.

Set pip timeout to 60 as the default 15 leads to occasional
CI failures due to timeouts.

Use '--parallel auto' in tox.ini [testenv:test] to run tests as
the latest crl.devutils only supports this way.

vereszol committed:
Use importlib instead of imp in setup.py

'imp' is deprecated, changing to avoid warnings whenever running
setup.py.

vereszol committed:
Use yaml.safe_load to open yaml target files

Recent pyyaml versions require loader when calling yaml.load() and
usage of yaml.safe_load() is recommended unless arbitrary object
deserialisation is needed.

Fixes #10

vereszol committed:
Fix pylint errors and warnings

Replace deprecated configparser readfp with read_file.

Fixes #15

OneTorcha commented on src/crl/rfcli/rfcli.py:
Can you add at the same the change to execute()

        for key, value in self.new_environment_variables.items():
            newpaths = value.split(os.pathsep)
            for path in newpaths:
                if not any(fnmatch.fnmatch(p, path) for p in sys.path):
                    sys.path.append(path)
        return run_cli(self.commandline[1:], exit=False)  # pylint: disable=unsubscriptable-object

this pythonpathsetter add_path() function is gone.

for path in newpaths:
                pythonpathsetter.add_path(path, True)

https://github.com/robotframework/robotframework/blob/master/src/robot/pythonpathsetter.py

vereszol replied:
Yes, that shall be added together with support for newer Robot Framework versions, as I pinned it down to <4.0 in 8fcb9b9 to get tox tests passing.

vereszol replied:
Done in 094f2a57c39e9

vereszol committed:
Drop support for jybot; use robot instead pybot

Robot Framework 3.1 removed 'pybot' and 'jybot' start-up scripts,
should use 'robot'.

vereszol committed:
Set language='en' in sphinx conf.py

Otherwise sphinx raises a warning.

vereszol committed:
Build and upload py3 wheel package

setup.cfg is changed to build py3 wheel and
upload both sdist and bdist_wheel to PyPI index

vereszol committed:
Add flake8 tests to tox.ini

vereszol committed:
Fix test for --rfcli-no-pythonpath

Message ofr import error changed in Robot 5.x, using regex to
accept both old and new message format.

vereszol committed:
Adapt for missing robot.pythonpathsetter.add_path

Robot Framework 6.1 removed add_path() from pythonpathsetter,
use sys.path.append instead.

Fixes #12

vereszol committed:
Add support for Python 3.8-3.12 and Robot 3.1-7.x

Update tox.ini env configuration to test all supported versions.

vereszol committed:
Prepare for releasing version 2.0.0

OneTorcha reviewed (approved):
Looks perfect.

vereszol committed:
Add Robot Framework :: Tool to classifiers

theodore86 commented:
bump
